pkg/store/consul/consulutil/ops.go

```diff
@@ -96,6 +96,42 @@ func List(
 	}
 }
 
+type ConsulKeyser interface {
+	Keys(prefix, separator string, q *api.QueryOptions) ([]string, *api.QueryMeta, error)
+}
+
+type keysReply struct {
+	keys      []string
+	queryMeta *api.QueryMeta
+	err       error
+}
+
+// SafeKeys performs a KV Keys operation that can be canceled. When the "done"
+// channel is closed, CanceledError will be immediately returned. (The HTTP RPC
+// can't be canceled, but it will be ignored.) Errors from Consul will be
+// wrapped in a KVError value.
+func SafeKeys(
```

Comment:
I don't know if I'd be asking for too much if I ask for more specificity in the name, like `CancelableKeys`?

I suppose all I'll ask for is consistency. If we have other uses of `Safe` that all mean "cancelable", then I suppose this will be OK. But if `Safe` means other things in other instances in our codebase, then I will be suspicious of this name.